### PR TITLE
Add type annotation to sequenceExpr

### PR DESCRIPTION
**Pull request by FBroy**
# Description

Add type annotation to a few other AST nodes.

## Changes:
* Add type annotation to sequenceExpr.
* Add type annotation to set operators.
    - unionOp
    - interceptOp
    - exceptOp
* Add type annotation to stringConcatenateOp
* Add type annotation to rangeSequenceExpr

# Deletes Source Branch?

No:     Work is (partially) done, but more work will be done on the source branch.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

File: src/typeInference/annotateRangeSequenceOperator.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { SequenceMultiplicity, SequenceType, ValueType } from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+
+export function annotateRangeSequenceOperator(
+	ast: IAST,
+	start: SequenceType,
+	end: SequenceType
+): SequenceType {
+	if (!start || !end) return undefined;
```
Comment (HectorPeeters) [inline src/typeInference/annotateRangeSequenceOperator.ts]: Can't we just always return an integer?

**Review by HectorPeeters — COMMENTED** [review]
Looks good to me!

**Review by MTimmerije — APPROVED** [review]
Seems good to me <3 ;)

**Review by HectorPeeters — APPROVED** [review]
Looks good!

**Review by maanpalle — APPROVED** [review]
Looks nice good job

**Pull request merged**